src/Rules/PHPUnit/AssertSameWithCountRule.php

```diff
@@ -44,6 +45,21 @@ public function processNode(Node $node, Scope $scope): array
 			];
 		}
 
+		if (
+			$right instanceof Node\Expr\MethodCall
+			&& $right->name instanceof Node\Identifier
+			&& strtolower($right->name->toString()) === 'count'
+			&& count($right->args) === 0
+		) {
+			$type = $scope->getType($right->var);
+
+			if ($type instanceof ObjectType && $type->isInstanceOf(\Countable::class)->yes()) {
```

Comment:
There should be a different condition: `(new ObjectType(\Countable::class))->isSuperTypeOf($type)->yes()`. It accounts for some special cases, like `StaticType`, and I like to be consistent.

Reply:
@ondrejmirtes Done :)